numpyro/distributions/distribution.py

```diff
@@ -129,24 +130,49 @@ def logpdf(self, x, *args, **kwargs):
 
 
 class jax_discrete(jax_generic, osp_stats.rv_discrete):
-    def _support_mask(self, k):
-        return (k >= self.a) & (k <= self.b) & (np.floor(k) == k)
+    def __new__(cls, *args, **kwargs):
+        return super(jax_discrete, cls).__new__(cls)
+
+    def __init__(self, *args, **kwargs):
+        self.is_logits = kwargs.pop("is_logits", False)
+        super(jax_discrete, self).__init__(*args, **kwargs)
+
+    def freeze(self, *args, **kwargs):
+        self._ctor_param.update(is_logits=kwargs.pop("is_logits", False))
+        return super(jax_discrete, self).freeze(*args, **kwargs)
 
-    # TODO: This will not work for the case that loc != 0
     def _support(self, *args, **kwargs):
```

Comment:
For my understanding, does `dist.binom.support(p=.., n=..)` also work or is this only meant to be used with frozen distributions?

Reply:
The support property is only used in frozen distribution, where all parameters are known: loc, scale, n, p,... I guess using `dist.binorm._support(p=..., n=...)` also works.